openedx/features/caliper_tracking/tests/tests.py

```diff
@@ -0,0 +1,50 @@
+"""
+
+"""
+
+import json
+import os
+from datetime import datetime
+
+from django.test import TestCase
+import logging
+
+from openedx.features.caliper_tracking.base_transformer import CaliperBaseTransformer
+from openedx.features.caliper_tracking.caliper_config import EVENT_MAPPING
+
+
+TEST_DIR_PATH = 'openedx/features/caliper_tracking/tests/'
+
+
+class CaliperTransformationTestCase(TestCase):
+    """Data driven test case that tests all transformers for expected output"""
+
+    maxDiff = None
+
+    def test_caliper_transformers(self):
+        test_files = [file for file in os.listdir(
+            '{}current/'.format(TEST_DIR_PATH)) if file.endswith(".json")]
+
+        for file in test_files:
+            input_file = '{}current/{}'.format(
+                TEST_DIR_PATH,
+                file
+            )
+            output_file = '{}expected/{}'.format(
+                TEST_DIR_PATH,
+                file
+            )
+
+            with open(input_file) as current, open(output_file) as expected:
+                event = json.loads(current.read())
+                expected_event = json.loads(expected.read())
+
+                expected_event.pop('id')
+
+                caliper_event = CaliperBaseTransformer(event).transform_event()
+                related_function = EVENT_MAPPING[event.get('event_type')]
+                caliper_event = related_function(event, caliper_event)
+
+                caliper_event.pop('id')
```

Comment:
Because `id` is a randomly generated uuid, so the `id` generated by the transformer will never be equal to the saved `id` in the expected output. That's why I don't believe we need to compare them.